tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+	ordererConfig.General.LocalMSPDir = "/etc/hyperledger/fabric/artifacts/msp"
+	ordererConfig.FileLedger.Location = "/shared/data"
+	if nsConfig.Metrics {
+		ordererConfig.Operations.TLS.Enabled = false
+		ordererConfig.Metrics.Provider = "prometheus"
+	}
+	return ordererConfig, nil
+}
+
+func (k8s K8s) dockerImage(component, dockerOrg, dockerTag, componentImage string) string {
+	var image string
+	if dockerOrg != "" {
+		image = fmt.Sprintf("%s/fabric-%s:%s", dockerOrg, component, dockerTag)
+	} else if componentImage != "" {
+		image = componentImage
+	}
+	return image
+}
+
+//GenerateConfigurationFiles - to generate all the configuration files
+func (k8s K8s) GenerateConfigurationFiles(upgrade bool) error {
+
+	network := nl.Network{TemplatesDir: paths.TemplateFilePath("k8s")}
+	err := network.GenerateConfigurationFiles(upgrade)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func (k K8s) dataPersistenceFilePath(config networkspec.Config) string {
+func (k8s K8s) launchK8sNetwork(config networkspec.Config, clientset *kubernetes.Clientset) error {
 
-	var path string
-	currDir, err := paths.GetCurrentDir()
+	launchConfig, err := k8s.k8sLaunchObject(config)
 	if err != nil {
-		logger.ERROR("Failed to get the current working directory")
+		logger.ERROR("Failed to launch the fabric k8s components")
+		return err
 	}
-	switch config.K8s.DataPersistence {
-	case "local":
-		path = paths.JoinPath(currDir, "alpine.yaml")
-	default:
-		path = paths.ConfigFilePath("pvc")
+	for i := 0; i < len(launchConfig); i++ {
+		err = k8s.CreateStatefulset(launchConfig[i], config, clientset)
+		if err != nil {
+			logger.ERROR("Failed to launch the fabric k8s network")
+			return err
+		}
 	}
-	return path
+	return nil
 }
 
-func (k K8s) deleteConfigMaps(numComponents int, componentType, orgName, tls, k8sType string) error {
+func (k8s K8s) buildClientset(kubeconfig *string) (*kubernetes.Clientset, error) {
 
-	componentsList := []string{fmt.Sprintf("%s-ca", orgName), fmt.Sprintf("%s-admincerts", orgName)}
-	var componentName string
-	for j := 0; j < numComponents; j++ {
-		componentName = fmt.Sprintf("%s%d-%s", componentType, j, orgName)
-		componentsList = append(componentsList, []string{fmt.Sprintf("%s-tls", componentName), fmt.Sprintf("%s-msp", componentName)}...)
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		logger.ERROR("Failed to create config for kubernetes")
+		return &kubernetes.Clientset{}, err
 	}
-	input := []string{"delete", k8sType}
-	input = append(input, componentsList...)
-	k.Arguments = input
-	_, err := networkclient.ExecuteK8sCommand(k.Args(), true)
+	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		return err
+		logger.ERROR("Failed to create clientset for kubernetes")
+		return &kubernetes.Clientset{}, err
 	}
-	return nil
+	return clientset, nil
 }
 
-func (k K8s) K8sNetwork(action string) error {
+//K8sNetwork --
+func (k8s K8s) K8sNetwork(action string) error {
```

Comment:
Stutter, just call this function `Network`

Reply:
fixed